internal/services/notification/action/runwebhookdelivery.go

```diff
@@ -72,3 +73,35 @@ func (h *ActionHandler) GetProjectRunWebhookDeliveries(ctx context.Context, req
 		HasMore:              hasMore,
 	}, nil
 }
+
+func (h *ActionHandler) RunWebhookRedelivery(ctx context.Context, runWebhookDeliveryID string) error {
+	err := h.d.Do(ctx, func(tx *sql.Tx) error {
+		var err error
+		runWebhookDelivery, err := h.d.GetRunWebhookDeliveryByID(tx, runWebhookDeliveryID)
+		if err != nil {
+			return errors.WithStack(err)
+		}
+		if runWebhookDelivery == nil {
+			return util.NewAPIError(util.ErrNotExist, errors.Errorf("runWebhookDelivery %q doesn't exist", runWebhookDeliveryID))
+		}
+
+		if runWebhookDelivery.DeliveryStatus == types.DeliveryStatusNotDelivered {
+			return util.NewAPIError(util.ErrBadRequest, errors.Errorf("can't redelivery a not delivered runwebhookdelivery"))
```

Comment:
Is this really necessary? One user could always create tons of new delivery when the state is delivered. Or you should also check that only one redelivery can be created for a specific webhoo.

Reply:
So you choosed to let the user create tons of redeliveries?

Reply:
I'll prefer to let the user create a new redelivery only if the previous redelivery has already been delivered with any outcome (ok or failed).